client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -352,10 +359,14 @@ export default connect(
 				dispatch( fetchReceipt( receiptId ) );
 			},
 			fetchSitePlans( site ) {
-				dispatch( fetchSitePlans( site.ID ) );
+				if ( site ) {
+					dispatch( fetchSitePlans( site.ID ) );
+				}
 			},
 			refreshSitePlans( site ) {
-				dispatch( refreshSitePlans( site.ID ) );
+				if ( site ) {
```

Comment:
This is pretty minor, but we already have a couple conditions above for refreshing site plans. I think it'd be a little clearer if we just added this to those conditions (or created a `shouldRefreshSitePlans`) instead of creating an additional one. Otherwise I think I'd read code like:

```js
if ( this.props.receipt.hasLoadedFromServer && this.hasPlanOrDomainProduct() ) {
  this.props.refreshSitePlans( this.props.selectedSite );
}
```

and assume that we're refreshing the plans for that site. With the addition of this condition, I have to be aware of how the `refreshSitePlans` prop works as well, as it is no longer just an action dispatcher.